modules/iam__enum_roles/main.py

```diff
@@ -124,7 +153,48 @@ def main(args, pacu_main):
 
                     data['roles_assumed'].append(role)
 
-    return data
+
+def main(args, pacu_main):
+    args = parser.parse_args(args)
+    iam: 'mypy_boto3_iam.IAMClient' = pacu_main.get_boto3_client('iam')
+
+    if args.role_name:
+        role_name = args.role_name
+        resp = iam.get_role(RoleName=role_name)
+        orig_trust_doc = json.dumps(resp['Role']['AssumeRolePolicyDocument'])
+    else:
+        role_name = f"PacuIamEnumRoles-{''.join(random.choice(chars) for _ in range(5))}"
+        iam.create_role(
+            RoleName=role_name,
+            Description="Created by Pacu for IAM Role enumeration.",
+            AssumeRolePolicyDocument='''
+            {
+                "Version": "2012-10-17",
+                "Statement": [{
+                    "Effect": "Deny",
+                    "Principal": {"AWS": "*"},
+                    "Action": "sts:AssumeRole"
+                }]
+            }
+            '''.strip(),
+        )
+
+    try:
+        run(args, role_name, pacu_main, iam)
+    except Exception as e:
+        raise e
+    finally:
+        if args.role_name:
+            print(f"Reverting the {role_name} trust policy.")
+            try:
+                iam.update_assume_role_policy(RoleName=role_name, PolicyDocument=orig_trust_doc)
+            except Exception as err:
+                print(f"unable to revert {role_name} to it's original state. this can be done manually by updating the assume"
+                      f"role document to the following: \n\n{orig_trust_doc}\n\n")
+                print(err)
+        else:
+            print(f"Cleaning up the {role_name} role.")
+            iam.delete_role(RoleName=role_name)
```

Comment:
Revert the trust policy doc if args.role_name was passed otherwise delete it since it's a temporary role created for this run.